data-model/src/main/resources/META-INF/native-image/io.micronaut.data/data-model/native-image.properties

```diff
@@ -1,17 +0,0 @@
-#
-# Copyright 2017-2021 original authors
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-# https://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-Args = --initialize-at-build-time=io.micronaut.data.model.query.builder.sql.Dialect,io.micronaut.data.model.DataType,io.micronaut.data.annotation.Join$Type
```

Comment:
With graalvm 21 this is not needed, but is failing without it in graalvm 17 (which is used currently in github).

Reply:
We shouldn't generate enums in metadata anymore. Where does it see them?

Reply:
In graalvm 17 it is failing and suggesting ot trace `--trace-class-initialization=io.micronaut.data.annotation.Join$Type` which tells that `Dialect` is referring `Join.Type` here 
```
public enum Dialect {
    /**
     * H2 database.
     */
    H2(true, false,
        EnumSet.of(
            DEFAULT,
            LEFT,
            LEFT_FETCH,
            RIGHT,
            RIGHT_FETCH,
            FETCH,
            INNER
        )),
```
in graalvm 21 this is not needed.

Reply:
Where does it see the dialect?

Reply:
It is showing long stack trace
```
Error: Classes that should be initialized at run time got initialized during image building:
 io.micronaut.data.model.query.builder.sql.Dialect was unintentionally initialized at build time. io.micronaut.data.model.query.builder.sql.Dialect caused initialization of this class with the following trace: 
	at io.micronaut.data.model.query.builder.sql.Dialect.<clinit>(Dialect.java:43)
Error: Classes that should be initialized at run time got initialized during image building:

	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Unknown Source)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:568)
	at java.lang.Class.getEnumConstantsShared(Class.java:3837)
	at java.lang.Class.enumConstantDirectory(Class.java:3859)
	at java.lang.Enum.valueOf(Enum.java:267)
	at sun.reflect.annotation.AnnotationParser.parseEnumValue(AnnotationParser.java:479)
	at sun.reflect.annotation.AnnotationParser.parseMemberValue(AnnotationParser.java:344)
	at sun.reflect.annotation.AnnotationParser.parseAnnotation2(AnnotationParser.java:282)
	at sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:121)
	at sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:73)
	at java.lang.Class.createAnnotationData(Class.java:4068)
	at java.lang.Class.annotationData(Class.java:4057)
	at java.lang.Class.createAnnotationData(Class.java:4073)
	at java.lang.Class.annotationData(Class.java:4057)
	at java.lang.Class.getAnnotation(Class.java:3940)
------------------------------------------------------------------------------------------------------------------------	at java.lang.reflect.AnnotatedElement.isAnnotationPresent(AnnotatedElement.java:292)
	at java.lang.Class.isAnnotationPresent(Class.java:3950)
	at com.oracle.svm.hosted.heap.PodFeature.isPodClass(PodSupport.java:270)
	at com.oracle.svm.hosted.SVMHost.computeHubType(SVMHost.java:506)
	at com.oracle.svm.hosted.SVMHost.createHub(SVMHost.java:421)

	at com.oracle.svm.hosted.SVMHost.registerType(SVMHost.java:268)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.createType(AnalysisUniverse.java:310)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.lookupAllowUnresolved(AnalysisUniverse.java:220)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.lookup(AnalysisUniverse.java:197)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.lookup(AnalysisUniverse.java:78)
	at com.oracle.graal.pointsto.infrastructure.WrappedConstantPool.lookupConstant(WrappedConstantPool.java:178)
	at org.graalvm.compiler.java.BytecodeParser.lookupConstant(BytecodeParser.java:4287)
	at org.graalvm.compiler.java.BytecodeParser.genLoadConstant(BytecodeParser.java:3960)
	at org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5167)
	at org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3406)
	at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.iterateBytecodesForBlock(SharedGraphBuilderPhase.java:712)
	at org.graalvm.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3366)
	at org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3208)
	at org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1134)
	at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.build(SharedGraphBuilderPhase.java:152)
	at org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1026)
	at org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:97)
	at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:114)
	at org.graalvm.compiler.phases.Phase.run(Phase.java:49)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:434)
	at org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
	at org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
	at com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:146)
	at com.oracle.graal.pointsto.meta.AnalysisMethod.parseGraph(AnalysisMethod.java:819)
	at com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsedHelper(AnalysisMethod.java:784)
	at com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:767)
	at com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.lookupEncodedGraph(InlineBeforeAnalysisGraphDecoder.java:120)
	at org.graalvm.compiler.replacements.PEGraphDecoder.doInline(PEGraphDecoder.java:1190)
	at org.graalvm.compiler.replacements.PEGraphDecoder.tryInline(PEGraphDecoder.java:1173)
	at org.graalvm.compiler.replacements.PEGraphDecoder.trySimplifyInvoke(PEGraphDecoder.java:1028)
	at org.graalvm.compiler.replacements.PEGraphDecoder.handleInvoke(PEGraphDecoder.java:982)
	at org.graalvm.compiler.nodes.GraphDecoder.processNextNode(GraphDecoder.java:871)
	at com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.processNextNode(InlineBeforeAnalysisGraphDecoder.java:186)
	at org.graalvm.compiler.nodes.GraphDecoder.decode(GraphDecoder.java:600)
	at org.graalvm.compiler.replacements.PEGraphDecoder.decode(PEGraphDecoder.java:854)
	at com.oracle.graal.pointsto.phases.InlineBeforeAnalysis.decodeGraph(InlineBeforeAnalysis.java:77)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:193)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:583)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:165)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureFlowsGraphCreated(MethodTypeFlow.java:152)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.getOrCreateMethodFlowsGraphInfo(MethodTypeFlow.java:110)
	at com.oracle.graal.pointsto.typestate.DefaultAnalysisPolicy.staticRootMethodGraph(DefaultAnalysisPolicy.java:199)
	at com.oracle.graal.pointsto.PointsToAnalysis.lambda$addRootMethod$0(PointsToAnalysis.java:315)
	at com.oracle.graal.pointsto.PointsToAnalysis$$Lambda$540/0x0000000401a133c0.run(Unknown Source)
	at com.oracle.graal.pointsto.PointsToAnalysis$2.run(PointsToAnalysis.java:493)
	at com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:187)
	at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:171)
	at com.oracle.graal.pointsto.util.CompletionExecutor$$Lambda$1065/0x0000000401a873e8.run(Unknown Source)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

Reply:
And line 43. of `Dialect` is this again 
```
H2(true, false,
        EnumSet.of(
            DEFAULT,
            LEFT,
            LEFT_FETCH,
            RIGHT,
            RIGHT_FETCH,
            FETCH,
            INNER
        )),
```

Reply:
Maybe we just need Join$Type entry? Not both

Reply:
Added back minimal changes to pass graalvm tests on graalvm 17 but we can delete them once we switch to 21.